lib/spack/spack/test/concretize.py

```diff
@@ -224,6 +224,15 @@ def concretize_multi_provider(self):
         s.concretize()
         assert s['mpi'].version == ver('1.10.3')
 
+    def concretize_difficult_packages(self):
+        """Test a couple of large package that are often broken due
+        to current limitations in the concretizer"""
+        s = Spec('dealii')
+        s.concretize()
+        assert s['dealii'].version == ver('9.0.1')
```

Comment:
Won't this test break every time a new version of `dealii` is added? Or is it running on a fake test package and not the real package?

You should also look into [pytest parametrization](https://docs.pytest.org/en/latest/parametrize.html) if you want to simplify this test a bit.

Reply:
> Won't this test break every time a new version of dealii is added?

It will, but we would see it immediately and I can update the test easily.

I will look at the parametrization and update the test accordingly. Thanks for the link.

Reply:
@adamjstewart I updated the test and also pinned the versions.